genesyscloud/resource_genesyscloud_flow.go

```diff
@@ -258,30 +145,20 @@ func updateFlow(ctx context.Context, d *schema.ResourceData, meta interface{}) d
 	flowID := ""
 
 	retryErr := withRetries(ctx, 16*time.Minute, func() *resource.RetryError {
-		// TODO: After public API endpoint is published and exposed to public, change to SDK method instead of direct invocation
-		path := architectAPI.Configuration.BasePath + "/api/v2/flows/jobs/" + jobId + "?expand=messages"
-		res := make(map[string]interface{})
 		// If possible, after changing to SDK method invocation, include correlationId we get earlier in this function when making the GET request
-		response, err := apiClient.CallAPI(path, "GET", nil, headerParams, nil, nil, "", nil)
+		flowJob, response, err := architectAPI.GetFlowsJob(jobId, []string{"messages"})
 		if err != nil {
 			// Nothing special to do here, but do avoid processing the response
 		} else if err == nil && response.Error != nil {
 			resource.NonRetryableError(fmt.Errorf("Error retrieving job status. JobID: %s, error: %s ", jobId, response.ErrorMessage))
```

Comment:
You can bring this line up into the `if err != nil` block and then remove this whole `else if err == nil && response.Error != nil`. We just do that check in the SDK code.